pkg/detectors/azuredirectmanagementkey/azuredirectmanagementkey.go

```diff
@@ -0,0 +1,163 @@
+var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	urlPat        = regexp.MustCompile(`https://([a-z0-9][a-z0-9-]{0,48}[a-z0-9])\.management\.azure-api\.net`)                                        // https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.APIM.Name/
+	keyPat        = regexp.MustCompile(detectors.PrefixRegex([]string{"azure", ".management.azure-api.net"}) + `([a-zA-Z0-9+\/]{83,85}[a-zA-Z0-9]==)`) // pattern for both Primary and secondary key
+
+	invalidHosts  = simple.NewCache[struct{}]()
+	noSuchHostErr = errors.New("no such host")
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{".management.azure-api.net"}
+}
+
+// FromData will find and optionally verify Azure Management API keys in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	logger := logContext.AddLogger(ctx).Logger().WithName("azuredirectmanagementkey")
+	dataStr := string(data)
+
+	urlMatchesUnique := make(map[string]string)
+	for _, urlMatch := range urlPat.FindAllStringSubmatch(dataStr, -1) {
+		urlMatchesUnique[urlMatch[0]] = urlMatch[1] // urlMatch[0] is the full url, urlMatch[1] is the service name
+	}
+	keyMatchesUnique := make(map[string]struct{})
+	for _, keyMatch := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		keyMatchesUnique[strings.TrimSpace(keyMatch[1])] = struct{}{}
+	}
+
+EndpointLoop:
+	for baseUrl, serviceName := range urlMatchesUnique {
+		for key := range keyMatchesUnique {
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_AzureDirectManagementKey,
+				Raw:          []byte(baseUrl),
+				RawV2:        []byte(baseUrl + ":" + key),
+			}
+
+			if verify {
+				if invalidHosts.Exists(baseUrl) {
+					logger.V(3).Info("Skipping invalid registry", "baseUrl", baseUrl)
+					continue EndpointLoop
+				}
+
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+
+				isVerified, verificationErr := s.verifyMatch(ctx, client, baseUrl, serviceName, key)
+				s1.Verified = isVerified
+				if verificationErr != nil {
+					if errors.Is(verificationErr, noSuchHostErr) {
+						invalidHosts.Set(baseUrl, struct{}{})
+						continue EndpointLoop
+					}
+					s1.SetVerificationError(verificationErr, baseUrl)
+				}
+			}
+
+			results = append(results, s1)
+		}
+	}
+
+	return results, nil
+}
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_AzureDirectManagementKey
+}
+
+func (s Scanner) Description() string {
+	return "Azure API Management provides a direct management REST API for performing operations on selected entities, such as users, groups, products, and subscriptions."
+}
+
+func (s Scanner) IsFalsePositive(_ detectors.Result) (bool, string) {
```

Comment:
If we want to skip some specific keyword can we implement it as [this](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/slackwebhook/slackwebhook.go#L117)